### ISSUE TITLE
3 forms of provenance for machine-readable statements

### ISSUE THREAD

**Issue by micheldumontier** [biolink/biolink-model#357]
I can see at least 3 different forms of provenance relating to a (machine-readable) statement e.g. one represented using the biolink model. 

The **first** form of provenance relates to the **procurement of the data file** containing the statement. In this case, there are potentially 4 different attributes of interest:  Where the data were obtained from, when the data were obtained, who obtained the data, and how they were obtained.

**provenance of (web-based) data retrieval**

- retrieved from [url]
- retrieved on [date] or retrieved over [time interval]
- retrieved by [agent]
- retrieved with [tool or more detailed process]

The **second** form of provenance relates to **source of the statement**. We'll assume here that the primary source of interest is in some machine-readable document (e.g. scientific article, pubmed abstract, data file).  There are at least 4 attributes of interest here: in what document can the statement be found, at what time, by whom, and what tool was used to record the assertion. 

**provenance of assertional reporting**

- reported in [document]
- reported on [date]
- reported by [agent]
- reported with [tool or more detailed process]

The **third** form of provenance relates to **how the machine-readable association was generated from the original document**. This really deals with any processing from that original document - for instance, some algorithm to identify protein-protein interactions from mass spectrometry experiments, human-led curation or automated text processing of the conclusion section of a scientific article into a structured, machine readable statement.

provenance related to data processing 

- generated from [document]
- generated on [date] or generated over [time interval]
- generated by [agent]
- generated with [tool or more detailed process]

**Comment by ehinderer** [comment]
One question for clarification: identifying the experimental method used in the original source (e.g. "predicted interaction by homology" versus "x-ray crystallographic characterization") would be accomplished by the _reported with_ statement under the **second form**?

**Comment by micheldumontier** [comment]
Correct. computational processing of data, whether directly produced by a machine or previously created data - in the broadest sense, falls in the third form. 

the manner in which we express what processing was done and how to represent it is cetainly the subject of further discussion, particularly because there are many valid views on how complicated it should be. The use of one or more controlled terminology such as [ECO](https://www.ebi.ac.uk/ols/ontologies/eco) could be one approach to indicate elements of the provenance in the generation of the statement. While ECO has been very useful for curating the biological scientific literature, I have specific concerns about the structure and organisation of ECO along with its scope and potential to extend beyond its current boundaries, but that's a different discussion. An alternative to a simple list of pre-defined terms is to use a terminology describe a sequence, tree, or graph of events, along with the participants and their roles. This is the approach of [PROV](https://www.w3.org/TR/prov-o/) and [SEPIO](https://github.com/monarch-initiative/SEPIO-ontology/wiki/The-SEPIO-Framework). The advantage here is that content creators have vastly more flexibility in crafting accurate, schema-compliant descriptions, but the disadvantage is that provenance processing tools must be able to process a provenance graph according to those specifications. That said, for the Translator community, it may make sense to adopt the flexible activity-based provenance description approaches that also admit the use of controlled vocabularies, like ECO, to help craft those machine-readable descriptions.

**Comment by ehinderer** [comment]
I've also read that ECO plans on collaborating with the Confidence Information Ontology for including information about confidence in addition to methodology in its annotations, but that's an ongoing effort. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6323956/
